app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -663,8 +662,12 @@ function getLocationEstimate()
         $locationInfo = Mage::getSingleton('core/session')->getLocationInfo();
 
         if (empty($locationInfo)) {
-            $locationInfo = $this->url_get_contents("http://freegeoip.net/json/".$this->getIpAddress());
-            Mage::getSingleton('core/session')->setLocationInfo($locationInfo);
+            //To receive the API results in the old freegeoip format, we need ipstack Access Key
+            $ipstackAccessKey = Mage::helper('core')->decrypt(Mage::getStoreConfig('payment/boltpay/ipstack_key'));
```

Comment:
I advice to move it to the Helper/Data.php
Like - getPublishableKeyOnePage()

It provides us an ability to use it in different places if needed.

Reply:
I feel like we only access this key from here

Reply:
I tend to lean more toward what @daisy1754 has said. 

 It is a matter of taste.  One may view a helper method as a simplification and clearer, reusable representation of the action which is being performed, and another may view it as code clutter, preferring to make simple one line call.

However, if reused, the extra decryption process does justify the helper method.